assets/app/view/game_card.rb

```diff
@@ -51,6 +72,8 @@ def render_header
           if owner?
           elsif user_in_game?(@user, @gdata)
             buttons << render_button('Leave', -> { leave_game(@gdata) })
+          elsif players.size < @gdata['max_players'] && @gdata['settings']['private_game']
```

Comment:
what about instead of joining the game, storing the state, then typing the password, you just type the password to join the game? that removes complexity and a step for the user